### PR TITLE
Make isolated a global argument

### PR DESCRIPTION
**Pull request by charliermarsh**
Closes https://github.com/astral-sh/uv/issues/3557.

### CODE REVIEW

File: crates/uv/src/cli.rs [hunk 1]
```diff
@@ -92,6 +87,11 @@ pub(crate) struct GlobalArgs {
 
     #[arg(global = true, long, overrides_with("preview"), hide = true)]
     pub(crate) no_preview: bool,
+
+    /// Avoid discovering a `pyproject.toml` or `uv.toml` file in the current directory or any
+    /// parent directories.
+    #[arg(global = true, long, hide = true)]
```
Comment (charliermarsh) [inline crates/uv/src/cli.rs:93]: We needed `global = true`.

**Review by zanieb — APPROVED** [review]
<3

**Comment by bluss** [comment]
the global options open the secret door where uv tells you about all you can really do. Don't know any other way to get it to spill the beans.

```
uv --preview
error: 'uv' requires a subcommand but one was not provided
  [subcommands: pip, venv, virtualenv, v, cache, self, clean, run, sync, lock,
version, generate-shell-completion, --generate-shell-completion, help]
```

**Comment by charliermarsh** [comment]
Hahaha the hidden subcommands?

**Comment by bluss** [comment]
Yep :slightly_smiling_face:  I guess, the shell completion is pretty open about them too though. Not `--generate-shell-completion` with dashes, that one is secret and `v` too.

**Comment by zanieb** [comment]
Funny that seems like a Clap bug.

They're not too secret though :) just not ready for primetime.

**Pull request merged**